Added .isEmpty() and .nonEmpty() methods to custom LibGDX collections

TCreutzenberg opened the pull request:
Pull request for issue https://github.com/libgdx/libgdx/issues/5196

NathanSweet commented:
`isEmpty` is ok. I'm not sure I like `nonEmpty`. The javadocs should match the rest, eg "Returns true if the array is empty." (ie "Returns true" and period). Using the class name can be OK, though it can also be clunky to repeat it many times. Using words like "array" or "heap" are find when the context makes it clear.

TCreutzenberg commented:
Thanks for the feedback!
I updated the comments and renamed .nonEmpty() to .hasItems().
Actually now it is a little clunky for maps and sets as they don't have "items" like arrays of lists but key-value pairs or only values.
It is highly desirable in my opinion to use the same naming for all of the collections so exchanging the type of collection does only need as few changes as possible.
If you're still unhappy with it I can also remove hasItems() completely. People could just use !isEmpty() then.

NathanSweet commented:
Thanks! Javadocs look good.

It's true `hasItems` isn't a good fit for maps. It seems OK for sets. Just `has`? `available`? Those aren't great either. Consistent method names would be nice if possible, but exchanging a list with a map will require more changes than just this. FWIW, we use `hasXxx` elsewhere, eg in scene2d `hasParent`, ` hasActions`.

Anyone else have ideas? Some options:

1. `hasItems` for all collections.
1. `hasItems` for collections with items (arrays, sets, etc) and `hasXxx` where it isn't a good fit (eg `hasEntries` for maps).
1. `nonEmpty` for all collections.
1. `!isEmpty()`.

kalexmills commented:
IMHO, Number 4 is best. isEmpty() already has a consistent meaning with its
counterpart defined in java.util.Collection. In any case,  hasItems() or
even nonEmpty() is just a convenience wrapper for !isEmpty(). We can
probably do without it, since the extra ! character is not very onerous to
type (carpal tunnel notwithstanding).

On Fri, May 4, 2018 at 11:33 AM Nathan Sweet <notifications@github.com>
wrote:

> Thanks! Javadocs look good.
>
> It's true hasItems isn't a good fit for maps. It seems OK for sets. Just
> has? available? Those aren't great either. Consistent method names would
> be nice if possible, but exchanging a list with a map will require more
> changes than just this. FWIW, we use hasXxx elsewhere, eg in scene2d
> hasParent, hasActions.
>
> Anyone else have ideas? Some options:
>
>    1. hasItems for all collections.
>    2. hasItems for collections with items (arrays, sets, etc) and hasXxx
>    where it isn't a good fit (eg hasEntries for maps).
>    3. nonEmpty for all collections.
>    4. !isEmpty().
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/libgdx/libgdx/pull/5199#issuecomment-386656840>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AVkoti44FS8_X4wJ0N3n91u5y5vrPn6Vks5tvILXgaJpZM4TlRjT>
> .
>
-- 

K. Alex Mills

TCreutzenberg commented:
Ok, one of the owners please tell me what to do ;)

NathanSweet commented:
Let's go with just `isEmpty` and not have a `nonEmpty` method.

TCreutzenberg commented:
Done.

NathanSweet commented:
Thanks!
